src/pretix/plugins/paypal2/static/pretixplugins/paypal2/pretix-paypal.js

```diff
@@ -73,12 +73,21 @@ var pretixpaypal = {
             pretixpaypal.locale = this.guessLocale();
         }
 
-        // If no payment option is selected, and we're not on the paypage (which doesn't have a continue button),
-        // disable the continue button
-        if (!pretixpaypal.paypage) {
-            if (!pretixpaypal.continue_button[0].form.elements['payment'].value) {
+        $("input[name=payment][value^='paypal']").change(function () {
+            if (pretixpaypal.paypal !== null) {
+                pretixpaypal.renderButton($(this).val());
+            } else {
                 pretixpaypal.continue_button.prop("disabled", true);
             }
+        });
+
+        $("input[name=payment]").not("[value^='paypal']").change(function () {
+            pretixpaypal.restore();
+        });
+
+        // If paypal is the only payment option, it is preselected and we must disable the continue button until the sdk is ready
```

Comment:
I think this comment is midleading as the selector also matches if there are multiple payment options and paypal is preselected – not sure, when this could happen, but maybe in some case of navigating between checkout-steps. The code itself is fine though as we should always disable the continue-button if paypal is selected. Also the bit about „until the sdk is ready“ could be a bit irritating when reading the code, as it kind of implies that once the SDK is ready, the button should be enabled again – which is not the case as it is only enabled in certain cases in renderButton and renderAPMs. Maybe update the comment to something like:

```suggestion
        // If paypal is pre-selected, we must disable the continue button and handle it after SDK is loaded
```